src/go/k8s/tests/e2e/2-redpanda-cluster/02-redpanda-cluster.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: redpanda.vectorized.io/v1alpha1
+kind: Cluster
+metadata:
+  name: cluster-sample-2
+  labels:
+    app.kubernetes.io/instance: "redpanda-cluster-sample"
```

Comment:
so the reason why the selection does not work for you is because you're providing the same labels and the selector is created from this.

Reply:
Agree. I think this PR addresses the issue by adding one more label from the statefulset.

Reply:
well I am not sure that's needed, you just need to name your second instance differently than the first one and I believe it should work without additional label

Reply:
I agree that having different instance label will not show this problem. My intention is to solve this corner case for the user.

Reply:
if user provides two same labels, that's user's error. Then it's obvious that the selectors for one will pick up the other as well. Right? Or am I missing something?

Reply:
I disagree this is a user error. If we can handle this gracefully, I would like to add this.